### PR TITLE
[11.0][MIG] purchase_picking_state

### PR DESCRIPTION
**Pull request by celm1990**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check Travis status

**Comment by celm1990** [comment]
@pedrobaeza travis status is OK, but what happened with codecov??

**Comment by pedrobaeza** [comment]
This is because the total test coverage has decreased. You have added code lines that are not tested.

**Comment by celm1990** [comment]
Sorry but I not understand, I no change code, only reorganize file purchase_view.xml into view's folder.
What test must be?

**Comment by pedrobaeza** [comment]
This Python file included in your migration is not tested: https://github.com/OCA/purchase-workflow/pull/546/files#diff-4a9632fc6c47e65d0ed3fc3d13f06503

**Comment by celm1990** [comment]
Not is new file, only move to models folder, nothing changes. Test is here https://github.com/OCA/purchase-workflow/pull/546/files#diff-8b775e6870ed76e28bb0a417361bd410R7

**Comment by pedrobaeza** [comment]
It's a new file in the repository. That's why Codecov complains, and the problem is that tests don't cover all the introduced code. See red lines in https://codecov.io/gh/OCA/purchase-workflow/compare/563bd9901068a2a95c1d3db1884dd184b5cd9563...a53fadb605a5e5ed8a212c34ed2f7b4b50e257f5/diff to see what is not tested.

**Comment by celm1990** [comment]
@pedrobaeza test done.

Now get error on travis, but is related to module stock_landed_cost_company_percentage.
image

Test on runbot and get same error.
image

Any Idea???

**Comment by pedrobaeza** [comment]
@bodedra your changes in https://github.com/OCA/purchase-workflow/commit/e75b0c416750965b0621e48dec1c7eeb2d384f9a seems to provoke this error. Can you please check?

**Comment by bodedra** [comment]
@pedrobaeza Yes. It will restrict user to confirm incoming shipment as our goal to create landed costs to every received items and auto validate it. We will review its workflow. Thank you.

CC @max3903 @jgrandguillaume @agyamuta @guewen

**Comment by pedrobaeza** [comment]
Then it's incorrect to have that check on test mode when testing other modules. You should use a technique similar to this one:

https://github.com/OCA/stock-logistics-workflow/blob/c601f3220bec5ea848329f170a8050ba177e87d5/stock_no_negative/models/stock_quant.py#L18-L24

**Comment by max3903** [comment]
@pedrobaeza @celm1990 Can you review #547 please?

**Comment by pedrobaeza** [comment]
@celm1990 Please rebase now that the patch has been merged.

**Comment by pedrobaeza** [comment]
@celm1990 you have merged the branch, not rebase it.

**Comment by celm1990** [comment]
image
 
Ok, must be merge.

**Comment by pedrobaeza** [comment]
Now, what you have perform is a merge operation somehow (maybe doing `git pull`):

seleccion_005

but you need to rebase your branch (assuming origin is OCA remote):
```
git fetch origin
git rebase origin/11.0
git push -f
```

**Comment by celm1990** [comment]
@pedrobaeza rebase is OK or not???

**Comment by pedrobaeza** [comment]
Yes, thanks

**Comment by celm1990** [comment]
@bodedra @max3903 can us review/approving PR???

**Comment by janbec** [comment]
@celm1990 
I tested and it seems like the `partially_received` status is not working. If only some parts of a purchase order have been delivered, the purchase order will still be in `not_received`. Once all goods are delivered, it will correctly display `done`.

**Comment by janbec** [comment]
@celm1990 
I just noticed I need to be more specific. The application works fine if you receive 1 out of 2 products of the same PO line.

If you have two PO lines and receive products for just one line, the status will be `not_received`.
image

**Comment by bealdav** [comment]
To add more value to this module check here https://github.com/OCA/purchase-workflow/issues/576

File: purchase_picking_state/i18n/vi_VN.po [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#. module: purchase_picking_state
+#: model:ir.model.fields,help:purchase_picking_state.field_purchase_order_picking_state
+msgid "Overall status based on all pickings"
+msgstr ""
+
+#. module: purchase_picking_state
+#: model:ir.model.fields,field_description:purchase_picking_state.field_purchase_order_picking_state
+msgid "Picking status"
+msgstr ""
```
Comment (congdpt) [inline purchase_picking_state/i18n/vi_VN.po:29]: You can set
```
msgstr "Trạng thái đơn hàng"
```

File: purchase_picking_state/models/purchase.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from odoo.tools.translate import _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.model
+    def get_picking_state(self):
+        return [
+            ('draft', ''),
```
Comment (congdpt) [inline purchase_picking_state/models/purchase.py:14]: I think you should change it to `('draft', _('Draft')),`

**Review by congdpt — COMMENTED** [review]
:+1:

**Comment by celm1990** [comment]
@janbec last commit fixed issue
@bealdav field picking_status is stored on DB

**Pull request closed without merging**